examples/speed_poc.cc

```diff
@@ -0,0 +1,243 @@
+#include <SDL2_mixer/SDL_mixer.h>
+#include <stdio.h>
+
+#include <vector>
+
+#include "core/include/sdl_handler.h"
+
+const int SIZE = 40;
+const int NUM_ROW = 15; // 800*600 => 600 / SIZE = 15
+const int NUM_COL = 50; // 800*600 => 800 / SIZE = 50
+
+class Camera {
+   public:
+    SDL_Rect viewport;
+    Camera(int width, int height) {
+        viewport.x = 0;
+        viewport.y = 0;
+        viewport.w = width;
+        viewport.h = height;
+    }
+
+    // update the camera position based on the player position
+    void update(int playerX, int playerY, int screenWidth, int screenHeight) {
+        viewport.x = playerX + SIZE / 2 - screenWidth / 2;
+        viewport.y = playerY + SIZE / 2 - screenHeight / 2;
+
+        // avoid camera to go out of the screen
+        if (viewport.x < 0)
+            viewport.x = 0;
+        if (viewport.y < 0)
+            viewport.y = 0;
+
+        // assume the screen is 800x600
+        if (viewport.x + viewport.w > NUM_COL * SIZE)
+            viewport.x = NUM_COL * SIZE - viewport.w;
+        if (viewport.y + viewport.h > NUM_ROW * SIZE)
+            viewport.y = NUM_ROW * SIZE - viewport.h;
+    }
+
+    // apply the camera to a rect
+    SDL_Rect apply(SDL_Rect& rect) const {
+        SDL_Rect relativeRect = rect;
+        relativeRect.x -= viewport.x;
+        relativeRect.y -= viewport.y;
+        return relativeRect;
+    }
+};
+
+class Block {
+   public:
+    int r, g, b;
+    Block(int x, int y, int r, int g, int b) {
+        this->rect.x = x;
+        this->rect.y = y;
+        this->rect.w = SIZE;
+        this->rect.h = SIZE;
+        this->r = r;
+        this->g = g;
+        this->b = b;
+        this->surface = sdl_handler::GetInstance().get_surface();
+    }
+
+    int get_x() { return this->rect.x; }
+    int get_y() { return this->rect.y; }
+
+    void draw(const Camera& camera) {
+        SDL_Rect relativeRect = camera.apply(this->rect);
+        SDL_SetRenderDrawColor(surface, this->r, this->g, this->b, 255);
+        SDL_RenderFillRect(surface, &relativeRect);
+    }
+
+   protected:
+    SDL_Rect rect;
+    SDL_Renderer* surface;
+};
+
+class Player : public Block {
+public:
+    // vx: speed in x direction
+    // ax: acceleration in x direction
+    float vx, ax; 
+
+    // max_speed: speed upper limit
+    float max_speed;
+
+    // is_sprinting: whether the player is sprinting
+    bool is_sprinting;
+
+    Player(int x, int y, int r, int g, int b) : Block(x, y, r, g, b), vx(0), ax(0), max_speed(300.0f), is_sprinting(false) {}
+
+    // Handle player input for movement and sprinting
+    void apply_input(SDL_Keycode key, bool is_pressed) {
+        float normal_acceleration = 200.0f;
```

Comment:
Maybe consider to put the constants to the private variable?

Reply:
good idea